WordPress/Classes/System/Root View/WordPressAuthenticatorProtocol.swift

```diff
@@ -8,20 +9,57 @@ protocol WordPressAuthenticatorProtocol {
 
 extension WordPressAuthenticator: WordPressAuthenticatorProtocol {
     static func loginUI() -> UIViewController? {
-        Self.loginUI(showCancel: false, restrictToWPCom: false, onLoginButtonTapped: nil, continueWithDotCom: { viewController in
-            guard Self.dotComWebLoginEnabled, let navigationController = viewController.navigationController else {
-                return false
-            }
+        Self.loginUI(
+            showCancel: false,
+            restrictToWPCom: false,
+            onLoginButtonTapped: nil,
+            continueWithDotCom: { viewController in
+                guard Self.dotComWebLoginEnabled, let navigationController = viewController.navigationController else {
+                    return false
+                }
 
-            Task { @MainActor in
-                let accountID = await WordPressDotComAuthenticator().signIn(from: navigationController, context: .default)
-                if accountID != nil {
-                    WordPressAppDelegate.shared?.presentDefaultAccountPrimarySite(from: navigationController)
+                Task { @MainActor in
+                    let accountID = await WordPressDotComAuthenticator().signIn(from: navigationController, context: .default)
+                    if accountID != nil {
+                        WordPressAppDelegate.shared?.presentDefaultAccountPrimarySite(from: navigationController)
+                    }
                 }
-            }
 
-            return true
-        })
+                return true
+            },
+            selfHostedSiteLogin: { viewController in
+                guard FeatureFlag.authenticateUsingApplicationPassword.enabled else { return false }
+                guard let window = viewController.view.window, let navigationController = viewController.navigationController else { return false }
```

Comment:
Does it need a window?

Reply:
The window will be used to present an AuthenticationService Safari view.